.github/ci/update_readme_dependencies.py

```diff
@@ -114,46 +289,60 @@ def check_bsp_readme(bsp_path) -> Any:
     except Exception:
         return
 
-    with open(readme_path, 'r+', newline='\n') as readme:
-        content = readme.readlines()
-        # Find or create 'Dependencies section' in the README.md
-        start_idx = index_regexp(content, DEPENDENCIES_START)
-        end_idx = index_regexp(content, DEPENDENCIES_END)
+    with open(readme_path, "r", encoding="utf-8") as f:
+        content = f.read()
 
-        if start_idx >= 0 or end_idx >= 0:
-            if set(table_md.splitlines()) == set([line[:-1] for line in content[start_idx + 1:end_idx]]):
-                return  # The table exists and is correct, we can return here
-            else:
-                print("[{}] Incorrect component table, updating...".format(str(bsp_path)))
-                del content[start_idx + 1:end_idx]
+    # UPDATE DEPENDENCIES TABLE
+    pattern = re.compile(
+        rf"{DEPENDENCIES_START}.*?{DEPENDENCIES_END}", re.DOTALL
+    )
+    new_section = f"{DEPENDENCIES_START}\n\n{table_capabilities_md}\n"
+    new_section += f"\n{DEPENDENCIES_END}"
+
+    if pattern.search(content):
+        updated_content = pattern.sub(new_section, content)
+        if content.strip() != updated_content.strip():
+            with open(readme_path, "w", encoding="utf-8") as f:
+                f.write(updated_content.strip())
+            print(f"{bsp_name} README.md capabilities updated")
         else:
-            print("[{}] First run on this file. Creating components section...".format(str(bsp_path)))
-            content.append(DEPENDENCIES_START + '\n')
-            content.append(DEPENDENCIES_END + '\n')
-            start_idx = index_regexp(content, DEPENDENCIES_START)
+            print(f"{bsp_name} README.md capabilities is up to date")
+    else:
+        print(f"Markers capabilities not found in {bsp_name} README.md")
 
-        # Write the generated table to the README.md
-        content.insert(start_idx + 1, table_md)
-        readme.truncate(0)
-        readme.seek(0, os.SEEK_SET)
-        readme.writelines(content)
-        # We have modified the README.md
-        raise Exception
+    # UPDATE EXAMPLES TABLE
+    pattern = re.compile(
+        rf"{EXAMPLES_START}.*?{EXAMPLES_END}", re.DOTALL
+    )
+    new_section = f"{EXAMPLES_START}\n\n{table_examples_md}\n"
+    new_section += f"\n{EXAMPLES_END}"
+
+    if pattern.search(content):
+        updated_content = pattern.sub(new_section, content)
+        if content.strip() != updated_content.strip():
+            with open(readme_path, "w", encoding="utf-8") as f:
+                f.write(updated_content.strip())
+            print(f"{bsp_name} README.md examples updated")
+        else:
+            print(f"{bsp_name} README.md examples is up to date")
+    else:
+        print(f"Markers examples not found in {bsp_name} README.md")
 
 
 def check_all_bsps(filenames):
-    ret = 0
+    # ret = 0
```

Comment:
Does the pre-commit hook work without this return value?

`__main__` of this python module returns `sys.exit(check_all_bsps(args.bsps))`. This was done so this module can return 0 or an error code that is evaluated by pre-commit

Reply:
This was only for debugging, when this is disabled, I can see all exceptions (specific line).